0_portfolio_input_check_functions.R

```diff
@@ -76,215 +76,216 @@ clean_portfolio_col_types <- function(portfolio, grouping_variables){
   }
-  
+
   portfolio
-  
+
 }
 
 add_holding_id <- function(portfolio){
   if (length(setdiff("holding_id", names(portfolio))) != 0) {
     portfolio$holding_id <- row.names(portfolio)
   }
-  
+
   portfolio
-  
+
 }
 
 check_missing_cols <- function(portfolio, grouping_variables){
-  
+
   required_input_cols <- c("holding_id", "market_value","currency","isin",grouping_variables, "number_of_shares")
-  
+
   if(!"number_of_shares" %in% colnames(portfolio)){portfolio$number_of_shares <- NA}
-  
-  
+
+
   missing_columns <-setdiff(required_input_cols,colnames(portfolio))
-  
+
   if(length(missing_columns) > 0){
     write_log(msg = paste0("The input file is missing the following data columns: ", missing_columns))
     stop(paste0("The input file is missing the following data columns: ", missing_columns))
   }
-  
+
   portfolio <- as_tibble(portfolio)
-  
+
   portfolio
-  
+
 }
 
 set_currency_timestamp <- function(currencies){
-  
+
   currencies <- currencies %>% mutate(ExchangeRate_2019Q2 = 1)
-  
+
   currencies <- currencies %>%
     select(Currency_abbr, paste0("ExchangeRate_",financial_timestamp)) %>%
     filter(!is.na(Currency_abbr), Currency_abbr != "") %>% distinct()
-  
+
   names(currencies)  <- c("currency", "exchange_rate")
-  
+
   currencies$exchange_rate <- as.numeric(currencies$exchange_rate)
-  
+
   currencies
 }
 
 ### Fin data cleaning functions
 map_security_sectors <- function(fin_data, sector_bridge){
-  
+
   initial_no_rows = nrow(fin_data)
-  
-  fin_data <- fin_data %>% left_join(sector_bridge %>% filter(source == "BICS") %>% select(-source), 
-                                     by = c("security_bics_subgroup" = "industry_classification")) %>% 
+
+  fin_data <- fin_data %>% left_join(sector_bridge %>% filter(source == "BICS") %>% select(-source),
+                                     by = c("security_bics_subgroup" = "industry_classification")) %>%
     mutate(security_icb_subsector = as.character(security_icb_subsector))
-  
+
   fin_data_na <- fin_data %>% filter(is.na(sector)) %>% select(-sector)
-  
+
   fin_data <- fin_data %>% filter(!is.na(sector))
-  
-  fin_data_na <- fin_data_na %>% left_join(sector_bridge %>% filter(source == "ICB")%>% select(-source), 
+
+  fin_data_na <- fin_data_na %>% left_join(sector_bridge %>% filter(source == "ICB")%>% select(-source),
                                            by = c("security_icb_subsector" = "industry_classification"))
-  
+
   fin_data <- fin_data %>% bind_rows(fin_data_na)
-  
+
   fin_data <- fin_data %>% select(-security_mapped_sector) %>% rename(security_mapped_sector = sector)
 
-  fin_data %>% group_by(security_mapped_sector) %>% filter(is.na(security_mapped_sector)) %>% summarise(count = n())
+  # FIXME: This seems to serve no purpose as it's not assigned to any variable
+  fin_data %>% group_by(security_mapped_sector) %>% filter(is.na(security_mapped_sector)) %>% summarise(count = n(), .groups = "drop")
```

Comment:
Should we remove this line or should we assign it to `fin_data`?

Reply:
Also, `.groups = "drop"` seems to be added recently. We want it, right?

Reply:
`.groups = "drop"` I believe is necessary with v1.0+ of `dplyr` to prevent a warning in the console